src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -382,6 +382,40 @@ uct_dc_mlx5_iface_dci_can_alloc(uct_dc_mlx5_iface_t *iface, uint8_t pool_index)
     return ucs_likely(pool->stack_top < iface->tx.ndci);
 }
 
+static UCS_F_ALWAYS_INLINE void
+uct_dc_mlx5_iface_dcis_array_copy(void *dst, void *src, size_t length)
```

Comment:
can we make src a "const void*" and change the definitions accordingly if needed?

Reply:
I see, we could assign the queue head from the old buffer to a temp local var on the stack
BTW, do we need to add ASAN macro to queue.h ?

Reply:
Wouldn't that still override the const modifier?
> BTW, do we need to add ASAN macro to queue.h ?

Yes we probably need

Reply:
> Wouldn't that still override the const modifier?

No because we don't really need to update the old buffer (make the old list empty). Using a local var as a copy of the head will allow the source buffer stay const because now only this local var will change

Reply:
gtest: `test_datatype, queue_splice` is expecting source length to be 0 after calling splice, can I just modify the test?